### PR TITLE
added Baudot ITA2 decoder

### LINKED ISSUE #355: Add Baudot decoder
https://en.wikipedia.org/wiki/Baudot_code
https://www.dcode.fr/baudot-code

### PR DESCRIPTION
**Pull request by Itamikame**
closes #355

### CODE REVIEW

File: ciphey/basemods/Decoders/baudot.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+from ciphey.iface import ParamSpec, Config, T, U, Decoder, registry
+
+@registry.register_multi((str, str), (bytes, bytes))
+class Baudot(Decoder[str, str]):
+    def decode(self, ctext: T) -> Optional[U]:
+        """Write the code that decodes here
+        ctext -> the input to the function
+        returns string
+    """
+        baudot_map = {
```
Comment (bee-san) [inline ciphey/basemods/Decoders/baudot.py:13]: This should go into CipheyDists! I'll need to write documentation for this :)

File: ciphey/basemods/Decoders/baudot.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+            '00111': 'U',
+            '11110': 'V',
+            '10011': 'W',
+            '11101': 'X',
+            '10101': 'Y',
+            '10001': 'Z',
+            '11011': '',
+            '11111': '',
+        }
+        baudot_digit_map = {
```
Comment (bee-san) [inline ciphey/basemods/Decoders/baudot.py:44]: ciphydists again

File: ciphey/basemods/Decoders/baudot.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+            '10011': '2',
+            '11101': '/',
+            '10101': '6',
+            '10001': '\"',
+            '11011': '',
+            '11111': '',
+        }
+
+        ret = ""
+        j = 0
```
Comment (bee-san) [inline ciphey/basemods/Decoders/baudot.py:77]: whats j? Can we get a better variable name? :)

**Review by bee-san — CHANGES_REQUESTED** [review]

File: ciphey/basemods/Decoders/baudot.py (hunk 1 above)
Comment (bee-san) [reply ciphey/basemods/Decoders/baudot.py:13]: You can always figure it out yourself :+1: https://github.com/Ciphey/CipheyDists/tree/master/cipheydists/translate

Convert your dictionaries into json files (with appropriate names) and add them to this folder in cipheydists.

And then we can easily call your dictionaries (see the other decoders for how to)

Comment (Itamai) [reply ciphey/basemods/Decoders/baudot.py]: i made a pull request for cipheydists can you check the changes i made?

File: ciphey/basemods/Decoders/baudot.py (hunk 3 above)
Comment (Itamai) [reply ciphey/basemods/Decoders/baudot.py]: my bad i forgot to rename it changed it to switch_to_digit_map for easier understanding of the code

**Comment by bee-san** [comment]
Hey! Your tests are failing :) This is because we had a bug in master branch (when you forked). We fixed this yesterday, if you merge with Master again on your fork it'll fix all the tests (hopefully) :partying_face:

**Comment by Itamikame** [comment]
The tests are now successful. I checked the log it was because the tests used the CipheyDists version without the baudot translation that one got added today to CipheyDists everything should be fine now 😃

**Comment by bee-san** [comment]
@all-contributors please add @Itamikame for code! :partying_face:

**Review by bee-san — APPROVED** [review]

**Comment by bee-san** [comment]
Hey we're changing from MIT to GPLv3 because some of our projects are licensed with that, do you approve? If you do not approve we will be forced to delete your code and rewrite it 😢 

By approving of this change, you are agreeing to re-license the code you wrote under GPLv3.

**Comment by Itamai** [comment]
> Hey we're changing from MIT to GPLv3 because some of our projects are licensed with that, do you approve? If you do not approve we will be forced to delete your code and rewrite it 😢
> 
> By approving of this change, you are agreeing to re-license the code you wrote under GPLv3.

I approve the license change you can use my code :)

**Pull request merged**